cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -273,16 +273,27 @@ const char* reverse_utf8(int64 context, const char* data, int32 data_len,
 
 // Truncates the string to given length
 FORCE_INLINE
-char* castVARCHAR_utf8_int64(int64 context, const char* data, int32 data_len,
-                             int64_t out_len, int32_t* out_length) {
-  // TODO: handle allocation failures
-  int32_t len = data_len <= static_cast<int32_t>(out_len) || out_len == 0
-                    ? data_len
-                    : static_cast<int32_t>(out_len);
-  char* ret = reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, len));
-  memcpy(ret, data, len);
-  *out_length = len;
-  return ret;
+const char* castVARCHAR_utf8_int64(int64 context, const char* data, int32 data_len,
+                                   int64_t out_len, int32_t* out_length) {
+  int32_t len = static_cast<int32_t>(out_len);
+  if (len >= data_len || len == 0) {
```

Comment:
It seems odd that `out_len == 0` results in a passthru rather than returning an empty string. Could you comment on that?

Reply:
I kept it as such to keep it consistent as the existing behavior

Reply:
@projjal @praveenbingo This was not an objection and nothing needs to be fixed; I was just curious about the rationale. Is this behavior a SQL convention I'm unaware of, for example? In any case, a comment explaining this would probably be useful